vermouth/processors/water_bias.py

```diff
@@ -0,0 +1,157 @@
+            a dict of secondary structure codes and
+            epsilon value for the water bias in kJ/mol
+        idr_regions:
+            regions defining the IDRs
+        prefix: str
+            prefix of the Go virtual-site atomtypes
+        system: :class:vermouth.System
+            the system of the molecules is used for
+            storing the nonbonded parameters
+        """
+        self.water_bias = water_bias
+        self.auto_bias = auto_bias
+        self.idr_regions = idr_regions
+        self.system = None
+
+    def assign_residue_water_bias(self, molecule, res_graph):
+        """
+        Assign the residue water bias for all residues
+        with a secondary structure element or that are
+        defined by the region selector. Region selectors
+        superceed the auto assignement.
+
+        Parameters
+        ----------
+        molecule: :class:vermouth.Molecule
+            the molecule
+        res_graph: :class:vermouth.Molecule
+            the residue graph of the molecule
+        """
+        for res_node in res_graph.nodes:
+            resid = res_graph.nodes[res_node]['resid']
+            chain = res_graph.nodes[res_node]['chain']
+            resname = res_graph.nodes[res_node]['resname']
+
+            if _in_region(resid, self.idr_regions):
+                eps = self.water_bias.get('idr', 0.0)
+            elif self.auto_bias:
+                sec_struc = res_graph.nodes[res_node]['cgsecstruct']
+                eps = self.water_bias.get(sec_struc, 0.0)
+            else:
+                continue
+
+            if eps == 0.0:
+                continue
+
+            vs_go_node = next(get_go_type_from_attributes(res_graph.nodes[res_node]['graph'],
+                                                          resid=resid,
+                                                          chain=chain,
+                                                          prefix=molecule.meta.get('moltype')))
+
+            # what is the blocks bb-type
+            bb_type = molecule.force_field.blocks[resname].nodes['BB']['atype']
+            size = _get_bead_size(bb_type)
+            # bead sizes are defined in the force-field file as
+            # regular, small and tiny
+            sigma = float(molecule.force_field.variables[size])
+            # update interaction parameters
+            atoms = (molecule.force_field.variables['water_type'], vs_go_node)
+            water_bias = NonbondParam(atoms=atoms,
+                                      sigma=sigma,
+                                      epsilon=eps,
+                                      meta={"comment": ["water bias", sec_struc]})
+            self.system.gmx_topology_params["nonbond_params"].append(water_bias)
+
+    def run_molecule(self, molecule):
+        """
+        Assign water bias for a single molecule
+        """
+        if not self.system:
+            raise IOError('This processor requires a system.')
```

Comment:
Can't we do `system = molecule.system`? Or do Molecules not have a reference to the system they're part of?

Reply:
I don't think they have. A molecule doesn't need to be part of a system anyways